src/App.tsx

```diff
@@ -1,24 +1,201 @@
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [loading, setLoading] = useState(false);
+  const [tempTodo, setTempTodo] = useState<null | Todo>(null);
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
+  const someCompleted = todos.some(todo => todo.completed);
+  const everyCompleted = todos.every(todo => todo.completed);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage('Unable to load a todo');
+      });
+  }, []);
+
+  const visibletodos = useMemo(
+    () => prepareTodos(todos, filteringField), [todos, filteringField],
+  );
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const todoCount = (currentTodos: Todo[]) => {
+    return currentTodos.filter(todo => !todo.completed).length;
+  };
+
+  const addTodo = (todo: Todo) => {
+    setLoading(true);
+    setTempTodo(todo);
+
+    return createTodo(todo)
+      .then(newTodo => {
+        setTodos(currentTodos => [...currentTodos, newTodo]);
+      })
+      .catch(() => setErrorMessage('Unable to add a todo'))
+      .finally(() => {
+        setLoading(false);
+        setTempTodo(null);
+      });
+  };
+
+  const deleteTodo = (todoId: number) => {
+    setLoadingTodoIds(ids => [...ids, todoId]);
+
+    return removeTodo(todoId)
+      .then(() => {
+        setTodos(currentTodos => currentTodos
+          .filter(todo => todo.id !== todoId));
+      })
+      .catch(() => setErrorMessage('Unable to delete a todo'))
+      .finally(() => setLoadingTodoIds(ids => ids.filter(id => id !== todoId)));
+  };
+
+  const updateTodo = (updatedTodo: Todo) => {
+    setLoadingTodoIds(currentIds => [...currentIds, updatedTodo.id]);
+
+    return updatingTodo(updatedTodo)
+      .then(todo => {
+        setTodos(currentTodos => {
+          const newTodos = [...currentTodos];
+          const index = newTodos.findIndex(
+            currentTodo => currentTodo.id === updatedTodo.id,
+          );
+
+          newTodos.splice(index, 1, todo);
+
+          return newTodos;
+        });
+      })
+      .catch(() => setErrorMessage('Unable to update a todo'))
+      .finally(() => setLoadingTodoIds(
+        currentIds => currentIds.filter(id => {
+          return id !== updatedTodo.id;
+        }),
+      ));
+  };
+
+  const deleteCompletedTodo = () => {
+    const completedIds = todos
+      .filter(todo => todo.completed)
+      .map(todo => todo.id);
+
+    setLoadingTodoIds(completedIds);
+
+    completedIds.forEach(id => deleteTodo(id));
+  };
+
+  const toggleAllHandler = () => {
+    if (everyCompleted) {
+      todos.forEach(
+        todo => updateTodo({ ...todo, completed: !todo.completed }),
+      );
+    } else {
+      const uncompletedTodos = todos.filter(todo => !todo.completed);
+
+      uncompletedTodos.forEach(
+        todo => updateTodo({ ...todo, completed: !todo.completed }),
+      );
+    }
```

Comment:
You can just go through all todos and change their status to the opposite of `everyCompleted`